src/Tasks/SystemState.cs

```diff
@@ -265,14 +239,107 @@ AssemblyTableInfo[] installedAssemblyTableInfos
         }
 
         /// <summary>
-        /// Serialize the contents of the class.
+        /// Writes the contents of this object out to the specified file.
+        /// TODO: once all derived classes from StateFileBase adopt new serialization, we shall consider to mode this into base class
+        /// </summary>
+        internal void SerializeCacheByTranslator(string stateFile, TaskLoggingHelper log)
+        {
+            try
+            {
+                if (!string.IsNullOrEmpty(stateFile))
```

Comment:
Wouldn't it be better to assert stateFile cannot be null via ErrorUtilities.Verify...NotNull instead of letting the method silently ignore it?

Reply:
It is consistent with current behavior, see former: https://github.com/dotnet/msbuild/blob/4b9f6122dfc8397ee0d939d7932df9cd5818c90c/src/Tasks/StateFileBase.cs#L37
I believe `stateFile` is optionable parameter of RAR and it sommision results in no cache persistence.

Reply:
Okay, I guess it's a dubious precedent, but a precedent nonetheless :)